pages/api/socketio-init.ts

```diff
@@ -0,0 +1,58 @@
+import axios from "axios"
+import { BehaviorSubject, defer, from, iif, timer } from "rxjs"
+import { map, switchMapTo } from "rxjs/operators"
+import { Server } from "socket.io"
+
+const { IEX_API_TOKEN: TOKEN } = process.env
+
+const sub = new BehaviorSubject<string>("")
+
+const ioHandler = (_req: any, res: any) => {
+  if (!res.socket.server.io) {
+    console.log("*First use, starting websocket")
+    const io = new Server(res.socket.server)
+
+    io.on("connection", (socket) => {
+      console.log("Client has connected")
+
+      socket.on("message", (symbol) => {
+        sub.next(symbol)
+      })
+
+      sub.subscribe((symbol) => {
+        iif(
```

Comment:
push into `pipe (filter (your conditional logic )` ABOVE your subscription, then keep piping.  you should only need 1 subscribe.

Reply:
Got it! The fact that I did not know that you can use ```pipe``` to extract the value out of a ```Subject``` demonstrates my lack of understanding of the fundamental concepts of RxJS.